src/Discovery/CollaboraDiscovery.php

```diff
@@ -12,35 +12,38 @@
 
 declare(strict_types=1);
 
-namespace Drupal\collabora_online\Cool;
+namespace Drupal\collabora_online\Discovery;
 
 use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Cache\MemoryCache\MemoryCacheInterface;
 use Symfony\Component\ErrorHandler\ErrorHandler;
 
 /**
  * Service to get values from the discovery.xml.
  */
 class CollaboraDiscovery implements CollaboraDiscoveryInterface {
```

Comment:
In the end this service is a value object. Probably a factory would help in retrieving either a fresh or a cached instance of it, otherwise it gets cached in the service container and doesn't get really invalidated.
That's why I would not implement any static caching at the moment. We don't want to have to know here the cache metadata regarding how the xml was retrieved.

Reply:
Now the "factory" is the *Fetcher->getDiscovery().
The value object is no longer cached (as an object) but the raw xml is cached.